resource-managers/yarn/src/main/scala/org/apache/spark/scheduler/cluster/YarnClusterSchedulerBackend.scala

```diff
@@ -64,4 +66,42 @@ private[spark] class YarnClusterSchedulerBackend(
     }
     driverLogs
   }
+
+  override def getDriverAttributes: Option[Map[String, String]] = {
+    var attributes: Option[Map[String, String]] = None
+    try {
+      val yarnConf = new YarnConfiguration(sc.hadoopConfiguration)
+      val containerId = YarnSparkHadoopUtil.getContainerId
+      val clusterId: Option[String] = try {
+        Some(YarnConfiguration.getClusterId(yarnConf))
+      } catch {
+        case _: HadoopIllegalArgumentException => None
+      }
+
+      val httpAddress = System.getenv(Environment.NM_HOST.name()) +
+        ":" + System.getenv(Environment.NM_HTTP_PORT.name())
+
+      // lookup appropriate http scheme for container log urls
+      val yarnHttpPolicy = yarnConf.get(
+        YarnConfiguration.YARN_HTTP_POLICY_KEY,
+        YarnConfiguration.YARN_HTTP_POLICY_DEFAULT
+      )
+      val user = Utils.getCurrentUserName()
+      val httpScheme = if (yarnHttpPolicy == "HTTPS_ONLY") "https://" else "http://"
+
+      attributes = Some(Map(
+        "HTTP_SCHEME" -> httpScheme,
+        "NODE_HTTP_ADDRESS" -> httpAddress,
+        "CLUSTER_ID" -> clusterId.getOrElse(""),
+        "CONTAINER_ID" -> ConverterUtils.toString(containerId),
+        "USER" -> user,
+        "LOG_FILES" -> "stderr,stdout"
+      ))
+    } catch {
+      case e: Exception =>
+        logInfo("Error while retrieving attributes on driver, so driver logs will not " +
+          "be replaced with custom log pattern", e)
+    }
+    attributes
```

Comment:
you can avoid the `var` here by just having the final expression be the result at the end of the try and the catch:

```scala
try {
  ...
  Some(Map(
    ...
  ))
} catch {
  case e: Exception =>
   logInfo(...)
   None
}
```